packages/flutter/test/widgets/scroll_aware_image_provider_test.dart

```diff
@@ -0,0 +1,335 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+import '../painting/image_test_utils.dart';
+import '../painting/mocks_for_image_cache.dart' show TestImage;
+
+void main() {
+  tearDown(() {
+    imageCache.clear();
+  });
+
+  RecordingPhysics _findPhysics(WidgetTester tester) {
+    return Scrollable.of(find.byType(TestWidget).evaluate().first).position.physics as RecordingPhysics;
+  }
+
+  ScrollMetrics _findMetrics(WidgetTester tester) {
+    return Scrollable.of(find.byType(TestWidget).evaluate().first).position;
+  }
+
+  testWidgets('DisposableBuildContext asserts on disposed state', (WidgetTester tester) async {
```

Comment:
Maybe move the DisposableBuildContext tests into their own file since the impl has its own home as well.